libcnb-cargo/src/cli.rs

```diff
@@ -1,35 +1,39 @@
-use clap::{App, AppSettings, Arg, SubCommand};
+use clap::{Parser, Subcommand};
 
-pub(crate) fn setup_cli_parsing<'a, 'b>() -> clap::App<'a, 'b> {
-    App::new(env!("CARGO_PKG_NAME"))
-        .bin_name("cargo")
-        .version(env!("CARGO_PKG_VERSION"))
-        .about(env!("CARGO_PKG_DESCRIPTION"))
-        .setting(AppSettings::GlobalVersion)
-        .setting(AppSettings::SubcommandRequiredElseHelp)
-        .subcommand(
-            SubCommand::with_name("libcnb")
-                .about("Allows working with buildpacks written with libcnb.rs")
-                .setting(AppSettings::SubcommandRequiredElseHelp)
-                .subcommand(
-                    SubCommand::with_name("package")
-                        .about("Packages a libcnb.rs Cargo project as a Cloud Native Buildpack")
-                        .arg(
-                            Arg::with_name("release")
-                                .long("release")
-                                .help("Build in release mode, with optimizations"),
-                        )
-                        .arg(
-                            Arg::with_name("target")
-                                .long("target")
-                                .default_value("x86_64-unknown-linux-musl")
-                                .help("Build for the target triple"),
-                        )
-                        .arg(
-                            Arg::with_name("no-cross-compile-assistance")
-                                .long("no-cross-compile-assistance")
-                                .help("Disable cross-compile assistance"),
-                        ),
-                ),
-        )
+#[derive(Parser)]
+#[clap(bin_name = "cargo")]
+pub(crate) enum Cli {
+    #[clap(subcommand)]
+    Libcnb(LibcnbSubcommand),
+}
+
+#[derive(Subcommand)]
+#[clap(version, about, long_about = None)]
+pub(crate) enum LibcnbSubcommand {
+    /// Packages a libcnb.rs Cargo project as a Cloud Native Buildpack
+    Package(PackageArgs),
+}
+
+#[derive(Parser)]
+pub(crate) struct PackageArgs {
+    /// Disable cross-compile assistance
+    #[clap(long)]
+    pub no_cross_compile_assistance: bool,
+    /// Build in release mode, with optimizations
+    #[clap(long)]
+    pub release: bool,
+    /// Build for the target triple
+    #[clap(long, default_value = "x86_64-unknown-linux-musl")]
+    pub target: String,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use clap::CommandFactory;
+
+    #[test]
+    fn verify_command() {
+        Cli::command().debug_assert();
```

Comment:
Nit: I had to look up [what this does](https://docs.rs/clap/3.1.1/clap/struct.App.html#method.debug_assert). The code alone left me puzzled. Maybe worth adding a comment?

Reply:
9664a46a80b0aed434abe5489c9c2220510a03bc